src/utils/userIP.ts

```diff
@@ -58,14 +66,29 @@ export function processUserIP(req: Request): string | null {
         return parseIP(firstEntry.for);
       } catch (e) {
         logger.warn(
-          `Request received with invalid content in "${HEADER_FORWARDED}" header.`
+          `Request received with invalid content in "${HEADER_FORWARDED}" header.`,
         );
         return null;
       }
     }
     case IPSources.XForwardedFor: {
-      logger.trace(`Sourcing User IP from "${HEADER_X_FORWARDED}" header.`);
-      return req.ip ?? null;
+      try {
+        logger.trace(`Sourcing User IP from "${HEADER_X_FORWARDED}" header.`);
+        if (isAPIGatewayProxyEvent(req)) {
+          const header = req.headers[HEADER_X_FORWARDED];
+          if (!header) return null;
+          const ip = header.split(",")[0];
+          return ip;
```

Comment:
Sorry I've missed a requirement here. Can we parse this `ip` with `parseIP` please? 🙏 We want to strip out things like a port number if they happen to be included.

Reply:
changed.